### PR TITLE
Add performance profiling to flare command

### PR DESCRIPTION
**Pull request by bmikaili**
### What does this PR do?

Changes the flare command logic for the core agent to zip in the flare process instead of the agent process. The agent process only collects the necessary files and writes them to disk. This allows us to avoid a timeout during the creation of a 120s CPU and heap profile during flare creation.

### Motivation

It would be useful to have performance profiling data for some support cases.

### Additional Notes

These changes only apply for Agent Core, there should be separate PRs to include this for other agents.

Also a limitation:  Now that the zip creation is separate from the creation of the flare files, I can't defer the removal of the temporary directory where the flare files are created before zipping. That means that theoretically, if the agent crashes during flare creation, the temp directory would not get cleaned up. Would like to hear opinions if there's a better way to make that safe. 

### Describe your test plan

1. Test that the flag works correctly for true and false
2. Test that profile files get created after 120s when creating a flare with the flag enabled
3. Test that flare didn't get broken on other platforms/windows.

### Notes for QA

Should be tested for every platform the core agent runs on.

### CODE REVIEW

File: releasenotes/notes/flare-add-performance-profiling-bbae32a7ca98eadc.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+enhancements:
+  - |
+    Changes flare creation so that archiving is handled from the flare process, leaving the agent to
```
Comment (ogaca-dd) [inline releasenotes/notes/flare-add-performance-profiling-bbae32a7ca98eadc.yaml:10]: IMO, enhancements should only describe the new feature (a flare contains profiling data) and no the technical details.

**Review by ogaca-dd — COMMENTED** [review]
As discussed in private, let's see if it is a good idea to:
- Create the profiling data from the agent that run `flare` command.
- When creating the flare, pass a list of files that must be included in the flare (profiling data in our case).

File: cmd/agent/app/flare.go [hunk 2]
```diff
@@ -81,42 +92,77 @@ func makeFlare(caseID string) error {
...
+	if err != nil {
+		return err
+	}
+
+	err = writeCPUProfile(tempDir, hostname, "cpu.pprof", cpuProfURL)
+	if err != nil {
+		return err
+	}
+
+	err = flare.WriteHTTPCallContent(tempDir, hostname, "heap_profile.log", heapProfURL)
```
[the remark this answers is not in the document]
Comment (bmikaili) [reply cmd/agent/app/flare.go:157]: Yeah I realized it does. I thought `RedactingWriter` opens and writes to file when it exists, and doesn't replace it completely.

File: cmd/agent/api/agent/agent.go [hunk 3]
```diff
@@ -86,16 +87,26 @@ func makeFlare(w http.ResponseWriter, r *http.Request) {
...
-	filePath, err := flare.CreateArchive(false, common.GetDistPath(), common.PyChecksPath, logFile)
-	if err != nil || filePath == "" {
+	tempDir, hostname, err := flare.CreateArchive(false, common.GetDistPath(), common.PyChecksPath, logFile)
+
+	if err != nil {
+		log.Errorf("Error creating flare directory: " + err.Error())
+		return
+	}
+
+	if err != nil || len(tempDir) == 0 {
```
[the remark this answers is not in the document]
Comment (bmikaili) [reply cmd/agent/api/agent/agent.go:97]: Right, The check above is redundant.

File: releasenotes/notes/flare-add-performance-profiling-bbae32a7ca98eadc.yaml (hunk 1 above)
Comment (bmikaili) [reply releasenotes/notes/flare-add-performance-profiling-bbae32a7ca98eadc.yaml:10]: Makes sense.

File: cmd/agent/app/flare.go [hunk 4]
```diff
@@ -32,6 +42,7 @@ func init() {
 	flareCmd.Flags().StringVarP(&customerEmail, "email", "e", "", "Your email")
 	flareCmd.Flags().BoolVarP(&autoconfirm, "send", "s", false, "Automatically send flare (don't prompt for confirmation)")
 	flareCmd.Flags().BoolVarP(&forceLocal, "local", "l", false, "Force the creation of the flare by the command line instead of the agent process (useful when running in a containerized env)")
+	flareCmd.Flags().BoolVarP(&enableProfiling, "profile", "p", false, "Add performance enableProfiling data to the flare. If used, the flare command will wait for 120s to collect enableProfiling data.")
```
[the remark this answers is not in the document]
Comment (bmikaili) [reply cmd/agent/app/flare.go:45]: Makes sense. I'll add that when the core functionality works.

**Comment by bmikaili** [comment]
Closed in favor of https://github.com/DataDog/datadog-agent/pull/6045

**Pull request closed without merging**